plugins/basic/nexus-repository-raw/src/main/java/org/sonatype/nexus/repository/raw/internal/RawContentPayloadMarshaller.java

```diff
@@ -57,10 +59,18 @@ public DateTime getLastUpdated() {
     };
   }
 
-  public static Payload toPayload(RawContent content) throws IOException {
+  public static Payload toPayload(final RawContent content) throws IOException {
     checkNotNull(content);
-    return new StreamPayload(content.openInputStream(),
+    return new StreamPayload(
+        new InputStreamSupplier() {
+          @NotNull
```

Comment:
The change in the nuget format above uses `@Nonnull` whereas this uses `@NotNull`...